eventcontent.go

```diff
@@ -319,7 +319,9 @@ func NewPowerLevelContentFromAuthEvents(authEvents AuthEventProvider, creatorUse
 	c.Defaults()
 	// If there is no power level event then the creator gets level 100
 	// https://github.com/matrix-org/synapse/blob/v0.18.5/synapse/api/auth.py#L569
-	c.Users = map[string]int64{creatorUserID: 100}
+	// If we want users to be able to set PLs > 100 with power_level_content_override
+	// then we need to set the upper bound: maximum allowable JSON value is (2^53)-1.
+	c.Users = map[string]int64{creatorUserID: 9007199254740991}
```

Comment:
Synapse still sets 100: https://github.com/matrix-org/synapse/blob/5c9e39e6192e952ba8a5bb8e5485bc6067f91699/synapse/event_auth.py#L756

... but that only works because they have special cases for no power level existing, which we don't: https://github.com/matrix-org/synapse/blob/5c9e39e6192e952ba8a5bb8e5485bc6067f91699/synapse/event_auth.py#L713-L715

Reply:
Right, and we don't want to do that? Also, will this work fine on WASM etc? I seem to recall large numbers making it fall over.

Reply:
I think it's just two different ways of solving the problem. In our case, we don't specifically check the existence of the previous power level event at the point of checking the power levels, we do it at the point that we set up the `Allower`. If no previous event exists, we just populate some sane restrictive defaults and everything after that point has no idea the event was ever missing.

Honestly I prefer our approach. It is more explicit about the fact that the defaults still apply.

It should work fine with WASM because it is defined as `int64` and not `int`.